File: DsWorkJava/dswork.core/dswork/core/mybatis/dialect/DB2Dialect.java
package dswork.core.mybatis.dialect;

/**
 * Dialect for DB2
 * @author skey
 */
public class DB2Dialect extends Dialect
{
	/**
	 * 是否支持分页，limit和offset
	 * @return boolean
	 */
	public boolean supportsLimitOffset()
	{
		return true;
	}

	/**
	 * 将sql变成分页sql
	 * @param sql 原始sql语句
	 * @param offset 跳过行数
	 * @param limit 返回行数
	 * @return String
	 */
	public String getLimitString(String sql, int offset, int limit)
	{
		StringBuilder sb = new StringBuilder(sql.length() + 110);
		sb.append("select * from ( select ROW_NUMBER() over() rn,  t_.* from (")
		.append(sql)
		.append(") t_ ")
		.append(") n_ where n_.rn>").append(offset).append(" and n_.rn<=").append(offset + limit);
		return sb.toString();
	}
}

File: DsWorkNet/Dswork.Core/Mybaits/Dialect/DB2Dialect.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Dswork.Core.Mybaits.Dialect
{
	/// <summary>
    /// Dialect for DB2
	/// </summary>
    public class DB2Dialect : Dialect
	{
		/// <summary>
		/// 是否支持分页，limit和offset
		/// </summary>
		/// <returns>Boolean</returns>
		public override Boolean SupportsLimitOffset()
		{
			return true;
		}

		/// <summary>
		/// 将sql变成分页sql
		/// </summary>
		/// <param name="sql">原始sql语句</param>
		/// <param name="offset">跳过行数</param>
		/// <param name="limit">返回行数</param>
		/// <returns>String</returns>
		public override String GetLimitString(String sql, int offset, int limit)
		{

            StringBuilder sb = new StringBuilder(sql.Length + 110);
            sb.Append("select * from ( select ROW_NUMBER() over() rn, _t.* from (")
            .Append(sql)
            .Append(") _t ")
            .Append(") _n where _n.rn > ").Append(offset).Append(" and _n.rn <=").Append(offset + limit);
            return sb.ToString();
		}
	}
}
